components/scream/src/physics/p3/atmosphere_microphysics.hpp

```diff
@@ -77,13 +77,16 @@ class P3Microphysics : public AtmosphereProcess
         const Spack& pmid_pack(pmid(icol,ipack));
         const Spack& T_atm_pack(T_atm(icol,ipack));
         const Spack& cld_frac_t_pack(cld_frac_t(icol,ipack));
+        // Ensure that the pseudo_density and pmid used by P3 is dry
+        pdel_dry(icol,ipack) = pseudo_density(icol,ipack) * (1-qv(icol,ipack));
+        pmid_dry(icol,ipack) = pmid_pack * (1-qv(icol,ipack));
```

Comment:
pmid_dry in general cannot simply be multiplied by (1-qv); rather, one has to do a scan over pdel_dry.

Reply:
By the way, if your wiring plan was simply to add two or three fields -- pmiddry, pdeldry, and possibly pmiddry -- for general use by any parameterization in homme_post_process, I support that. That's exactly what EAM does. In particular, it might be natural to add these to HommeDynamics::update_pressure.

Reply:
> pmid_dry in general cannot simply be multiplied by (1-qv); rather, one has to do a scan over pdel_dry.

Yes, I agree that dry pressure at a given level needs should be the weight of all the dry pressure thicknesses above it. As written, all levels except for the one computed are still wet pressure thickness. I feel dumb for saying, but I'm not sure how to implement the needed cumulative sum on packs which are already inside a parallel for loop... There's also the issue that pdel gives the mass between top and bottom of a cell so cumsum over pdel gives interface pressures which then need to be converted to midlevel pressures...

Reply:
@PeterCaldwell see e.g. HommeDynamics::update_pressure re: the loop mechanics; and note that I suggest adding pmiddry, pdeldry, and possibly pintdry to that routine, following EAM's example.

Reply:
@ambrad , thanks for catching that.  That was an oversight on my part.  In reference to what is done in EAM, I see this line https://github.com/E3SM-Project/scream/blob/master/components/eam/src/physics/cam/physics_types.F90#L1472 which appears to be where EAM sets the dry coordinates.  It also appears to be making the same mistake.  Can we say then that EAM is not doing things correctly either?  If so, I should probably change this in EAM but that would be an upstream PR.

Reply:
oh wait, nevermind, I didn't pay attention to the next few lines. :-)